### ISSUE TITLE
1.2.0-rc ipset fw rule not applied

### ISSUE THREAD

**Issue by working-name** [hestiacp/hestiacp#877]
### In a few words, please describe the issue that you're experiencing:
Attempting to add an ipset and a rule for it seems to fail.

### What steps did you take when the issue occured? 
1. Log into the Hestia Control Panel using Safari, add ipset, choose china one
2. Save - takes a really long time
3. Add rule to deny tcp connections from ipset - takes a really long time
4. `iptables -nvL` is not reconciled with control panel rules; ipset rule is missing in iptables

### Expected behavior:
apply ipset rule

### Does this issue always occur, or is it intermittent?
No idea, I only tried this once.

### Operating system distribution and release:
Ubuntu 20.04 LTS

### Which version of Hestia Control Panel is currently installed?
1.2.0-rc

### Which branch are you using?
master

### When did this issue occur? After a clean installation, or after an upgrade?
after a clean install

### Which of the following web server configurations are you using?
`bash hst-install-ubuntu.sh -a no -v no`

### In order to better assist, please post any relevant log information below:
nothing about this in /var/log/hestia/error.log
/usr/local/hestia/system.log:
```
system.log:2020-06-27 14:36:36 v-add-firewall-rule  'DROP' 'ipset:test' '' 'TCP' 'blockchina'
```

### Related issues or forum threads:
couldn't find another issue, found some mildly connected comment here: https://forum.hestiacp.com/t/hestia-1-2-0-rc-we-need-you/1061/36 - but it seems ipset wasn't installed, not similar to my case

### Additional notes:
I'm not familiar with hestia, no idea where else it logs stuff. Let me know what you need.

**Comment by working-name** [comment]
Side note: since the page takes ages to load I'm thinking it might have to do with attempting to update china list? 

My suspicion was somewhat on the money - server could not resolve anything. I paused the ipset rule - no change. I deleted the ipset rule and the server could once again resolve domain names. 

No changes in `iptables -nvL` from before the failed rule and after its removal. 

I also tried directing requests to `@8.8.8.8` and it said no servers could be reached. OUTPUT chain had a policy of ACCEPT at all times. The one thing I didn't check on was if bind9 was running or not.

Attempting to reproduce the issue above by re-adding a fw rule using the ipset failed. Server can still resolve just fine, but the rule only exists in hestia not in iptables.

**Comment by Lupul** [comment]
Check if the ip lists were download `/usr/local/hestia/data/firewall/ipset/`
If they are, you can manually load whem with: `v-update-firewall-ipset`
Use `ipset list -t` to confirm the ipset list was loaded.

**Comment by working-name** [comment]
It was loaded prior to the DNS lookup failure, but the rule was nonexistent. Right now (after deleting the fw rule and re-adding it without touching the ipset) it seems to load fast again, rule appears in the CP but not in iptables output.

```py
# ls -la /usr/local/hestia/data/firewall/ipset/
total 96
drwxr-xr-x 2 root root  4096 Jun 28 00:10 .
drwxr-xr-x 3 root root  4096 Jun 28 23:06 ..
-rwxr-xr-x 1 root root  1951 Jun 27 14:19 blacklist.sh
-rw-rw---- 1 root root 83093 Jun 27 22:41 test.v4.iplist

# ipset list -t
Name: test
Type: hash:net
Revision: 6
Header: family inet hashsize 2048 maxelem 1048576
Size in memory: 138200
References: 0
Number of entries: 5336

# iptables -nvL > ip4.before
# v-update-firewall-ipset
# iptables -nvL > ip4.after
# diff -bi -C0 ip4.before ip4.after
*** ip4.before	2020-06-28 23:14:00.211463561 -0400
--- ip4.after	2020-06-28 23:14:08.075329074 -0400
***************
*** 5 ****
! 94089 9576K fail2ban-SSH  tcp  --  *      *       0.0.0.0/0            0.0.0.0/0            tcp dpt:22
--- 5 ----
! 94129 9579K fail2ban-SSH  tcp  --  *      *       0.0.0.0/0            0.0.0.0/0            tcp dpt:22
***************
*** 9 ****
!  3096  211K ACCEPT     all  --  *      *       127.0.0.1            0.0.0.0/0
--- 9 ----
!  3097  211K ACCEPT     all  --  *      *       127.0.0.1            0.0.0.0/0
***************
*** 11 ****
! 29054 1509K ACCEPT     tcp  --  *      *       0.0.0.0/0            0.0.0.0/0            multiport dports 80,443
--- 11 ----
! 29056 1509K ACCEPT     tcp  --  *      *       0.0.0.0/0            0.0.0.0/0            multiport dports 80,443
***************
*** 24 ****
! Chain OUTPUT (policy ACCEPT 39 packets, 5884 bytes)
--- 24 ----
! Chain OUTPUT (policy ACCEPT 92 packets, 14880 bytes)
***************
*** 205,206 ****
!    10   776 REJECT     all  --  *      *       46.254.94.217        0.0.0.0/0            reject-with icmp-port-unreachable
!    18  1448 REJECT     all  --  *      *       103.123.8.75         0.0.0.0/0            reject-with icmp-port-unreachable
--- 205,206 ----
!    14  1112 REJECT     all  --  *      *       46.254.94.217        0.0.0.0/0            reject-with icmp-port-unreachable
!    19  1500 REJECT     all  --  *      *       103.123.8.75         0.0.0.0/0            reject-with icmp-port-unreachable
***************
*** 208 ****
! 76759 8285K RETURN     all  --  *      *       0.0.0.0/0            0.0.0.0/0
--- 208 ----
! 76794 8287K RETURN     all  --  *      *       0.0.0.0/0            0.0.0.0/0
```

Essentially no changes in rules after running `v-update-firewall-ipset`.

```py
# iptables -nvL | grep -P 'ipset|test|china|HESTIA'
  458 84627 fail2ban-HESTIA  tcp  --  *      *       0.0.0.0/0            0.0.0.0/0            tcp dpt:8083
Chain fail2ban-HESTIA (1 references)
```
No ipset, or 'test' or 'china' in rules. It never gets added I guess.

**Comment by Lupul** [comment]
After creating the test list have you added a firewall rule that matches it? Should show up in `v-list-firewall`

`v-update-firewall` reload the firewall rules, but you should not have to execute it manually, after adding a new firewall rule it is  executed in the background

**Comment by working-name** [comment]
Okay `v-list-firewall` gave me a clearer picture. The rule was listed but the port column was missing. The GUI doesn't notify that `port` is required so I left it blank assuming it would be understood as `0-65535`. Also no logs hinted at this being an issue. Once I provided a port range the rule shows up in iptables as well.

Since then I also tried `*` or `all` or `any` for port which generate an unchiseled error message:
```py
{<br> "test": {<br> "IP_VERSION": "v4",<br> "AUTOUPDATE": "yes",
<br> "SUSPENDED": "no",<br> "SOURCE": "http://ipverse.net/ipblocks/data/countries/cn.zone",
<br> "TIME": "00:10:02",<br> "DATE": "2020-06-29"<br> }<br>}
<br>Error: invalid port format :: all
```

Otherwise it's handling multiple ports and ranges conversion: `200,252,300-400` becomes `dports 200,252,300:400`
